okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -28,6 +29,135 @@ private DepResolution() {
+    return checkAvailable(testList, testList, false);
+  }
+
+  /**
+   * Test required dependencies for a set of modules against an existing set.
+   * @param available existing set of modules
+   * @param testList modules whose dependencies we want to check
+   * @param removeIfMissingDep remove from testList if dependency check fails
+   * @return empty string if all OK; error message otherwise
+   */
+  public static String checkAvailable(Collection<ModuleDescriptor> available,
+      Collection<ModuleDescriptor> testList, boolean removeIfMissingDep) {
+
+    Map<String, List<InterfaceDescriptor>> ints = getProvidedInterfaces(available);
+    List<String> list = new LinkedList<>();
+
+    Iterator<ModuleDescriptor> iterator = testList.iterator();
+    while (iterator.hasNext()) {
+      ModuleDescriptor md = iterator.next();
+      List<String> res = checkDependenciesInts(md, available, ints);
+      if (!res.isEmpty()) {
+        if (removeIfMissingDep) {
+          available.remove(md);
+          ints = getProvidedInterfaces(available);
+          iterator.remove();
+          iterator = testList.iterator();
+        } else {
+          list.addAll(res);
+        }
+      }
+    }
+    if (list.isEmpty()) {
+      return "";
+    } else {
+      return String.join("\n", list);
+    }
+  }
+
+  /**
+   * Test interfaces for a set of enabled modules.
+   * @param modsEnabled enabled modules
+   * @return list of errors ; empty list if not error(s)
+   */
+  public static List<String> checkEnabled(Map<String, ModuleDescriptor> modsEnabled) {
+    return interfaceCheck(modsEnabled, modsEnabled, Collections.emptyList(), false,
+        new HashSet<>());
+  }
+
+  /**
+   * Install modules with dependency checking only.
+   * @param modsAvailable available modules
+   * @param modsEnabled enabled modules (for some tenant)
+   * @param tml install list with actions
+   * @param reinstall whether to re-install
+   */
+  public static void install(Map<String, ModuleDescriptor> modsAvailable,
+      Map<String, ModuleDescriptor> modsEnabled, List<TenantModuleDescriptor> tml,
+      boolean reinstall) {
+    installMaxIterations(modsAvailable, modsEnabled, tml, reinstall, 100);
+  }
+
+  /**
+   * Return top-N set of modules - in order of module ID.
+   * @param limit max number for each module (Top-N)
+   * @param mdl modules to consider (will be modified!)
+   */
+  public static void getLatestProducts(int limit, List<ModuleDescriptor> mdl) {
+    mdl.sort(Collections.reverseOrder());
+    Iterator<ModuleDescriptor> it = mdl.listIterator();
+    String product = "";
+    int no = 0;
+    while (it.hasNext()) {
+      ModuleDescriptor md = it.next();
+      if (!product.equals(md.getProduct())) {
+        product = md.getProduct();
+        no = 0;
+      } else if (no >= limit) {
+        it.remove();
+      }
+      no++;
+    }
+  }
+
+  static boolean moduleDepProvided(List<ModuleDescriptor> modules, ModuleDescriptor md) {
```

Comment:
Is it intended that optional dependencies are checked is if they were required dependencies? If yes can this added to the javadoc?

Reply:
Added javadoc and changed the method a bit.